doc/admin-guide/files/remap.config.en.rst

```diff
@@ -518,20 +514,95 @@ mapping rules. (It is activated before any mappings and is never
+optimizaion. To control this for specific remap rules, a named filter called `ip_allow` is pre-defined. This named filter is
+activated implicitly in default. To stop applying the special rule, disable the `ip_allow` filter as shown below.
+
+::
 
-To allow control of :ref:`IP Allow<ip-allow>` it is treated as an implicitly active and named
-filter. When this filter is active IP Allow checks are done before remap. To prevent this for
-specific remap rules, this filter, named "ip_allow", must be disabled. The common way of doing this
-would be ::
+   # ip_allow.yaml
+   ip_allow:
+      - apply: in
+        ip_addrs: 198.51.100.0/24
+        action: deny
+        method: ALL
 
+   # remap.config
    .deactivatefilter ip_allow
    map ...
    map ...
    .activatefilter ip_allow
 
-Note this entirely disables IP Allow checks for those remap rules.
+Note this entirely disables :file:`ip_allow.yaml` checks for those remap rules.
+
+Evaluation Order and Matching Policy
+------------------------------------
+
+ATS evaluates multiple ACL filters in the following order:
+
+1. Special "deny all" filter in :file:`ip_allow.yaml`
+2. In-line Filter in :file:`remap.config`
+3. Named Filter in :file:`remap.config`
+4. Filters in :file:`ip_allow.yaml`
+
+When an ACL filter is found, ATS stops processing ACL filters depending on the mathcing policy configured by
+:ts:cv:`proxy.config.url_remap.acl_matching_policy`.
+
+First Explicit Match Wins Policy
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the default matching policy. ATS stops processing ACL filters only if an explicit match of IP address and HTTP method is
+found.
+
+::
+
+   # ip_allow.yaml
+   ip_allow:
+      - apply: in
+        ip_addrs: 198.51.100.0/24
+        action: deny
+        method: PUT
+
+   # remap.config
+   .definefilter named-filter-1 @action=deny @method=HEAD
+   .definefilter named-filter-2 @actio==allow @method=POST
+
+   .activatefilter named-filter-1
+   .activatefilter named-filter-2
+
+   map http://www.example.com/ http://internal.example.com/ @action=deny @method=GET
+
+The result of this example is below. The evaluation applied from left to right until explicit match is found.
+
+====== ================ ================ ================ ================ =============
+Method In-line Filter   Named Filter 1   Named Filter 2   ip_allow.yaml    result
+====== ================ ================ ================ ================ =============
+GET    deny  (explicit) \-               \-               \-               denied  (403)
+HEAD   allow (implicit) deny  (explicit) \-               \-               denied  (403)
+POST   allow (implicit) allow (implicit) allow (explicit) \-               allowed (200)
+PUT    allow (implicit) allow (implicit) deny  (implicit) deny  (explicit) denied  (403)
+DELETE allow (implicit) allow (implicit) deny  (implicit) allow (implicit) allowed (200)
+====== ================ ================ ================ ================ =============
```

Comment:
I really like these tables you made. This will be very helpful to users. I'm sorry if I come across as harping on the explicit/implicit terminology, but I don't prefer it here either, but for different reasons than the policy naming. For this "match on ip and method" policy we don't want to say, that the inline filter implicitly allows HEAD requests, for example. It does not in fact allow them, but rather leaves it to other filters or rules to decide. It's more accurate to say it doesn't apply to HEAD requests. How about this tweak?

```
====== ================ ================ ================ ================ =============
Method In-line Filter   Named Filter 1   Named Filter 2   ip_allow.yaml    result
====== ================ ================ ================ ================ =============
GET    deny             \-               \-               \-               denied  (403)
HEAD   N/A              deny             \-               \-               denied  (403)
POST   N/A              N/A              allow            \-               allowed (200)
PUT    N/A              N/A              N/A              deny             denied  (403)
DELETE N/A              N/A              N/A              allow            allowed (200)
====== ================ ================ ================ ================ =============
```

Reply:
PTAL